### DISCUSSION TITLE
Only getting price sensor data in HA from Pow-P1

### DISCUSSION THREAD

**Discussion by tfriberg** [UtilitechAS/amsreader-firmware#734] [Q&A]
Have I missed a step? No power sensors (only price, voltage and rssi) have been created by autodiscovery in HA.

In MQTT explorer it looks like all is well:

<img width="910" alt="image" src="https://github.com/UtilitechAS/amsreader-firmware/assets/867763/7de48f98-e7c7-4193-bd87-18ec4d2e1c32">

MQTT config in AMS reader UI:

<img width="762" alt="image" src="https://github.com/UtilitechAS/amsreader-firmware/assets/867763/d6b65901-4ecb-4b0a-a383-b30fd4ebfe92">

In HA I only get price, RSSI and Voltage from the AMS reader MQTT entity:

<img width="1272" alt="image" src="https://github.com/UtilitechAS/amsreader-firmware/assets/867763/cd394267-51df-4d36-9123-2f80d6b707f5">

<img width="429" alt="image" src="https://github.com/UtilitechAS/amsreader-firmware/assets/867763/436f5706-739a-4937-a041-de76e73fba86">

All looks good in AMs reader UI

<img width="822" alt="image" src="https://github.com/UtilitechAS/amsreader-firmware/assets/867763/e6ab7e2c-1742-4c7e-80f3-70c38dae91c8">

**Comment by ArnieO** [comment]
That is strange. Is the power sensor still not available?
We have many Home Assistant users, I've never heard about missing sensors before. 
I am not using HA myself, so I have to reach out to the community here:

**HELP NEEDED!**
**Is there anyone here that understand what might be going on in @tfriberg 's HA?**

**Comment by tfriberg** [comment]
I've also tried getting the power sensors to HA via [amshan-homeassistant](https://github.com/toreamun/amshan-homeassistant) using mqtt and `amsreader` as topic (as defined/described in my first post) but get a timeout error.

**Comment by tfriberg** [comment]
For some reason, after unplugging and replugging the reader, I now have the data I expected. No idea why though.

**Reply by ArnieO** [reply]
Strange - but glad to hear it is solved!
